packages/core/http/core-http-router-server-internal/src/response.ts

```diff
@@ -51,6 +52,11 @@ const redirectionResponseFactory: KibanaRedirectionResponseFactory = {
   redirected: (options: RedirectResponseOptions) => new KibanaResponse(302, options.body, options),
 };
 
+const notModifiedResponseFactory: KibanaNotModifiedResponseFactory = {
+  notModified: (options: HttpResponseOptions = {}) =>
+    new KibanaResponse(304, options.body, options),
+};
```

Comment:
NIT-pick, but I given returning a body payload with a `304` is against the specification, we probably want to ignore a potential body passed as option and always use `undefined` here, as we did for `204`

https://github.com/elastic/kibana/blob/dc927d98baa30c0e63a464ff02a2dca46ce251bb/packages/core/http/core-http-router-server-internal/src/response.ts#L47

Reply:
@pgayvallet thank you, that does make sense